src/Controller/RegistrationController.php

```diff
@@ -26,40 +25,37 @@ public function __construct(EmailVerifier $emailVerifier)
     }
 
     #[Route('/register', name: 'app_register')]
-    public function register(Request $request, UserPasswordHasherInterface $userPasswordHasher, EntityManagerInterface $entityManager): Response
+    public function register(Request $request, UserPasswordHasherInterface $userPasswordHasher, EntityManagerInterface $entityManager): JsonResponse
     {
-        $user = new User();
-        $form = $this->createForm(RegistrationFormType::class, $user);
-        $form->handleRequest($request);
+        try {
+            $user = new User();
+            $resultJson = json_decode($request->getContent(), true);
 
-        if ($form->isSubmitted() && $form->isValid()) {
-            // encode the plain password
+            $user->setEmail($resultJson["username"]);
             $user->setPassword(
                 $userPasswordHasher->hashPassword(
                     $user,
-                    $form->get('plainPassword')->getData()
+                    $resultJson["password"]
                 )
             );
+            $user->setRoles($resultJson["roles"]);
+
+            $userArray = get_object_vars($user);
+            if ( count($userArray) <= 3 ) {
+                return $this->json(['status' => '400', 'message' => 'Erro ao prosseguir com esse cadastro. Preencha todos os campos para prosseguir cadastro'], 200, ['Content-Type'=>'application/json; charset=utf-8']);
+            }
+            $userSearched = $entityManager->getRepository(User::class)->findUserByEmail($user->getEmail());
+            if ( $userSearched > 0 ) {
```

Comment:
Does it really work? You ain't returning a number from the `$userSearched = $entityManager->getRepository(User::class)->findUserByEmail($user->getEmail());`

Shouldn't you validate that this variable isn't empty ? 

EDIT: 

`//     * @return User[] Returns an array of User objects`

Reply:
Yes, I validate with empty value and worked correctly